tools/odbc/connect/connection_string_attributes.json

```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "keyword": "database",
+    "enum": "DATABASE",
+    "description": "The path to the database file.",
+    "options": "",
+    "example": "mydb.duckdb",
+    "default": "\":memory:\""
+  },
+  {
+    "keyword": "dsn",
+    "enum": "DSN",
+    "description": "The name of the DSN to connect to.",
+    "options": "",
+    "example": "duckdb",
+    "default": ""
+  },
+  {
+    "keyword": "allow_unsigned_extensions",
```

Comment:
Rather than hard-coding these options perhaps we can dynamically generate them from the options we expose in the `DBConfig`? You can get a list with `DBConfig::GetOptions` or `DBConfig::GetOptionNames`. 

In Python we allow all of these to be passed as options using the config, e.g.:

```py
import duckdb
duckdb.connect(':memory:', config={'default_null_order': 'nulls last'})
```

I can imagine we might want to allow this in the ODBC driver as well.